backend/backend.c

```diff
@@ -32,8 +32,10 @@ bool wlr_backend_start(struct wlr_backend *backend) {
 }
 
 void wlr_backend_destroy(struct wlr_backend *backend) {
-	if (backend->impl->destroy) {
+	if (backend->impl && backend->impl->destroy) {
```

Comment:
I am. This one is in the final cleanup commit, this "need" came from the last libinput types that do not have impl at all.